src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -284,4 +284,27 @@ List<Integer> findEodIdsForTransfer(ObjectRecordStatusEntity status, ExternalLoc
     )
     ExternalObjectDirectoryEntity findByIdsAndFailure(Integer mediaId, Integer caseDocumentId, Integer annotationDocumentId, Integer transcriptionDocumentId,
                                                       List<Integer> failureStatesList);
+
+    @Query(
+        """
+            SELECT COUNT(eo) > 0
+            FROM ExternalObjectDirectoryEntity eo
+            WHERE eo.status = :inboundStatus
+            AND eo.media = :media
+            AND eo.externalLocationType = :inboundLocation
+            AND eo.id NOT IN
+              (
+              SELECT eod.id FROM ExternalObjectDirectoryEntity eod, ExternalObjectDirectoryEntity eod2
+              WHERE eod.media = eod2.media
+              AND eod.status = :inboundStatus
+              AND eod.externalLocationType = :inboundLocation
+              AND eod2.status != :ignoredUnstructuredStatus
+              AND eod2.externalLocationType IN :destinationLocations
+              )
+            """
+    )
```

Comment:
i dont think inner query needs to have 2 tables in it, for instance, i think this is what you are after in native sql:-

```sql
SELECT *
FROM darts.external_object_directory eod
WHERE eod.ors_id = 2
--AND eo.media = :media
AND eod.elt_id = 1
and eod.med_id is not null
AND NOT exists
  (
  SELECT 1 FROM darts.external_object_directory eod2
  WHERE eod.med_id = eod2.med_id
  AND eod2.ors_id != 9
  AND eod2.elt_id IN (2,3)
  )
  ```

Reply:
you're right, will change this cheers